File: Swapity Swap/Swapity Swap.c
#include<stdio.h>

#define Hd main(void)
Hd
{
    int n,m,x,y,a,b;
    scanf("%d%d\n%d%d\n%d%d",&n,&m,&x,&y,&a,&b);
    int ar[n];
    for(int i=1;i<=n;i++){
        int aux = 1,pos=i;
        if(x<=pos && pos<=y)
            pos = (x+y)-pos;
        if(a<=pos&&pos<=b)
            pos = (a+b)-pos;
        while(pos!=i){
            aux++;
            if(x<=pos && pos<=y)
                pos = (x+y)-pos;
            if(a<=pos&&pos<=b)
                pos = (a+b)-pos;
        }
        int mn = m % aux;
        for(int j=0;j<mn;j++){
            if(x<=pos && pos<=y)
                pos = (x+y)-pos;
            if(a<=pos&&pos<=b)
                pos = (a+b)-pos;
        }
        ar[pos-1] = i;
    }
    for(int i=0;i<n;i++)printf("%d\n",ar[i]);
}

File: Swapity Swap/Swapity Swap.pas
program Hd;

var
    n, m, x, y, a, b, i, aux, pos, mn, j: Int64;
    ar: array of integer;

begin
    readln(n, m);
    readln(x, y);
    readln(a, b);
    
    SetLength(ar, n);
    
    for i := 1 to n do
    begin
        aux := 1;
        pos := i;
        
        if (x <= pos) and (pos <= y) then
            pos := (x + y) - pos;
        
        if (a <= pos) and (pos <= b) then
            pos := (a + b) - pos;
        
        while pos <> i do
        begin
            aux := aux + 1;
            
            if (x <= pos) and (pos <= y) then
                pos := (x + y) - pos;
            
            if (a <= pos) and (pos <= b) then
                pos := (a + b) - pos;
        end;
        
        mn := m mod aux;
        
        for j := 1 to mn do
        begin
            if (x <= pos) and (pos <= y) then
                pos := (x + y) - pos;
            
            if (a <= pos) and (pos <= b) then
                pos := (a + b) - pos;
        end;
        
        ar[pos - 1] := i;
    end;
    
    for i := 0 to n - 1 do
        writeln(ar[i]);
end.